### DISCUSSION TITLE
Lock the settings

### DISCUSSION THREAD

**Discussion by Joci78** [arendst/Tasmota#19455] [General]
Dear community,

Does Tasmota support any option for me to lock the settings? For example, if someone gets the ESP, they cannot read a password or fingerprint.

**Comment by s-hadinger** [comment]
This would be useless. You can easily dump the flash memory with the correct wiring.

**Reply by Joci78** [reply]
I need to place a sensor in a place where unauthorized people can access it. I don't want them to access the information. What is the suggestion for such a situation?

**Reply by s-hadinger** [reply]
Unfortunately there is no solution for you. Tasmota is designed for devices which you have physical control on.

ESP8266 has no way to secure the hardware. Period. Any electrical access to the chip gives you the ability to dump the flash and all the credentials.
ESP32 has some fuses, but we don't use them because once the ESP32 is locked, you can easily brick the device.

So don't put any sensitive credentials in your device in the wild.

**Reply by Joci78** [reply]
I have an ESP32 for this purpose. It would have been nice if Tasmota supported this.
Thank you very much for your answer.

**Reply by sfromis** [reply]
Tasmota does some mild scrambling of flash settings content (including Wifi credentials), meaning that it does take a bit of additional effort to read it, but certainly nothing like encryption or pretenses of security.

**Comment by Wintespe** [comment]
Of course, we all have the problem with WiFi accessible devices in the wild. This affects all devices not just Tasmota. Where can I find the "mild scrambling" in the source code of Tasmota to check the security of the current implementation? A little extra effort to read it is definitely appreciated.
Thnx

**Reply by arendst** [reply]
Can we stop this waste of time. If you want to know how it works dive in the source code.

The current 4k settings area is unscrambled in core. If saved from the gui it is xor with a fixed value. If a filesystem is present a hidden unscrambled file is also present. In addition a last known good unscrambled file too.

**Reply by Wintespe** [reply]
I just wanted to know the best place to start digging into the source code, no further discussion.

**Reply by sfromis** [reply]
The "mild scrambling" is when retrieving settings in dump format, and means **no security**. It is useless if you want to make life hard for an attacker with physical device access.
